src/utils/arkTransaction.ts

```diff
@@ -166,3 +168,159 @@ const nLocktimeMinSeconds = 500_000_000n;
+
+    for (let i = 0; i < tx.inputsLength; i++) {
+        const inp = tx.getInput(i);
+        if (!inp.witnessUtxo) {
+            throw new Error(`Input ${i} is missing witnessUtxo`);
+        }
+        prevoutScripts.push(inp.witnessUtxo.script);
+        prevoutAmounts.push(inp.witnessUtxo.amount);
+    }
+
+    // Verify tapScriptSig signatures
+    if (!input.tapScriptSig || input.tapScriptSig.length === 0) {
+        throw new Error(`Input ${inputIndex} is missing tapScriptSig`);
+    }
+
+    // Verify each signature in tapScriptSig
+    for (const [tapScriptSigData, signature] of input.tapScriptSig) {
+        const pubKey = tapScriptSigData.pubKey;
+        const pubKeyHex = hex.encode(pubKey);
+
+        // Skip verification for excluded pubkeys
+        if (excludePubkeys.includes(pubKeyHex)) {
+            continue;
+        }
+
+        // Extract sighash type from signature
+        // Schnorr signatures are 64 bytes, with optional 1-byte sighash appended
+        let sighashType = SigHash.DEFAULT;
+        let sig = signature;
+
+        if (signature.length === 65) {
+            // Last byte is the sighash type
+            sighashType = signature[64];
+            sig = signature.subarray(0, 64);
+        } else if (signature.length !== 64) {
+            throw new Error(
+                `Invalid signature length: ${signature.length} (expected 64 or 65)`
+            );
+        }
+
+        // Verify sighash type is allowed
+        if (!allowedSighashTypes.includes(sighashType)) {
+            const sighashName = `0x${sighashType.toString(16)}`;
+            const allowedNames = allowedSighashTypes
+                .map((t) => `0x${t.toString(16)}`)
+                .join(", ");
+            throw new Error(
+                `Invalid sighash type ${sighashName} for input ${inputIndex}, pubkey ${pubKeyHex}. Allowed: ${allowedNames}`
+            );
+        }
+
+        // Find the tapLeafScript that matches this signature's leafHash
+        if (!input.tapLeafScript || input.tapLeafScript.length === 0) {
+            throw new Error(
+                `Input ${inputIndex} is missing tapLeafScript for tapScriptSig verification`
+            );
+        }
+
+        // Search for the leaf that matches the leafHash in tapScriptSigData
+        const leafHash = tapScriptSigData.leafHash;
+        const leafHashHex = hex.encode(leafHash);
+        let matchingScript: Uint8Array | undefined;
+        let matchingVersion: number | undefined;
+
+        for (const [_, scriptWithVersion] of input.tapLeafScript) {
+            const script = scriptWithVersion.subarray(0, -1);
+            const version = scriptWithVersion[scriptWithVersion.length - 1];
+
+            // Compute the leaf hash for this script and compare as hex strings
+            const computedLeafHash = tapLeafHash(script, version);
+            const computedHex = hex.encode(
+                new Uint8Array(computedLeafHash as any)
+            );
```

Comment:
```suggestion
            const computedHex = hex.encode(computedLeafHash);
```
`computedLeafHash` is a `Bytes` = an alias of `Uint8Array`, shouldn't be needed to cast